code/cfile/cfilesystem.cpp

```diff
@@ -188,48 +189,96 @@ cf_root *cf_create_root()
 
 struct _file_list_t {
 	SCP_string name;
+	SCP_string sub_path;
 	time_t m_time;
 	size_t size;
+	int pathtype;
+	int offset;
+
+	_file_list_t() : m_time(0), size(0), pathtype(CF_TYPE_INVALID), offset(0) {}
 };
 
 static bool sort_file_list(const _file_list_t &a, const _file_list_t &b)
 {
+	if ( a.sub_path.empty() && !b.sub_path.empty() ) {
+		return stricmp(a.name.c_str(), b.sub_path.c_str()) < 0;
+	}
+
+	if ( !a.sub_path.empty() && b.sub_path.empty() ) {
+		return stricmp(a.sub_path.c_str(), b.name.c_str()) < 0;
+	}
+
+	if ( !a.sub_path.empty() && !b.sub_path.empty() ) {
+		int rc = stricmp(a.sub_path.c_str(), b.sub_path.c_str());
+
+		if (rc) {
+			return rc < 0;
+		}
```

Comment:
This is redundant, no?
``if(rc)`` is equivalent to ``if(rc != 0)``, and since ``rc < 0`` implies ``rc != 0``, you can just return ``rc < 0``, no?

Reply:
This is just a comparison of the paths so it needs to fall through to the filename comparison if the paths are equal.